### PR TITLE
feat: on-device bird ID with WingCLIP-0.3, 95.0 percent top-1 in the browser

### LINKED ISSUE #280: Background during identification on iOS, shows stripes on the side?
Too fast to capture

### LINKED ISSUE #283: On-device inference is single-threaded; cross-origin isolation unlocks threads but breaks ORT wasm loading
On-device inference currently runs single-threaded, on one of ten cores. The page isn't cross-origin isolated, so `SharedArrayBuffer` is unavailable and onnxruntime-web has no way to use threads.

Measured in the app page against the local dev server:

```
crossOriginIsolated: false
SharedArrayBuffer:   undefined
hardwareConcurrency: 10
```

Adding the headers to `public/_headers` does produce isolation, and `_headers` is honored by Workers static assets:

```
/*
  Cross-Origin-Opener-Policy: same-origin
  Cross-Origin-Embedder-Policy: credentialless
```

```
crossOriginIsolated: true
SharedArrayBuffer:   available
```

But that alone breaks identification. The upload flow failed 3/3, each time hitting the full 90s timeout with "Download failed: Failed to fetch" on the model gate.

It isn't the asset download. All four model assets still fetch byte-exact under COEP (14386199 / 25165824 / 8620924 / 24123497), including the clone + `cache.put` + `arrayBuffer` sequence `fetchCached` uses. And the error carried no URL, which after #279 means it didn't come from the asset fetch path at all.

The bundle ships `ort-wasm-simd-threaded-*.wasm`. Isolation is what makes ORT select the threaded build and spawn workers, and that's the path that fails. So the change that unlocks threads is the same change that breaks it, which is why this can't just be a headers commit.

I also expected COEP to break the cross-origin Wikimedia images and it didn't, so that's one less thing to worry about.

What's left is figuring out the worker/wasm loading under isolation, probably `ort.env.wasm.wasmPaths` plus whatever CORP the worker and wasm need, and confirming `numThreads` actually ends up above 1. Worth measuring the real speedup before committing to it. Going from 1 to ~4 threads on a 39M-param model could matter a lot for phones, or the wasm could already be memory-bound and it's a wash. No idea which until someone tries it.

Low priority: it works fine today, just slower than the hardware allows.

### PR DESCRIPTION
**Pull request by jlian**
Replaces server-side GPT bird identification with a model that runs entirely in the browser. No photo leaves the device, identification is free per request, and it works offline once the model is cached. Closes #260 

**This is the application half.** The training pipeline, parity harnesses, and research log that produced these artifacts land in a follow-up PR (#278). Nothing here imports them, so this stands alone.

## Accuracy

**95.00% absolute top-1** on the 3,322-photo validation split, using the full shipping pipeline.

There is no honest comparison against current production. The only GPT-5.4-mini measurement is 83-87% on **23 self-labelled images**, where a single photo moves the score by 4.3 points. No GPT baseline at scale exists or is planned. What can be compared is the same reranker across backbones on the same split:

| model | reranked top-1 | recall ceiling | headroom used |
|---|---|---|---|
| ViT-B 86.6M | 90.04 | 94.52 | 95.3% |
| **TinyCLIP 38.3M (this PR)** | **93.80** | **97.14** | **96.6%** |

Those use the pooled prior. Adding the month axis takes the shipping model to 95.00.

## The model

`WingCLIP-0.3` is a TinyCLIP-39M student distilled from **`WingCLIP-0.1`**, fine-tuned on ground truth, then WiSE-FT blended at alpha 0.60. BioCLIP-2 is the grandteacher: it taught 0.1, and 0.1 taught 0.3.

NABirds top-1, all 24,633 images:

| model | params | NABirds top-1 |
|---|---|---|
| BioCLIP-2 (grandteacher) | 428M | 86.41 |
| WingCLIP-0.1 (teacher) | 86.6M | 89.93 |
| **WingCLIP-0.3 (this PR)** | **38.3M** | **86.90** |

It **beats the grandteacher at 11x fewer parameters**, while sitting 3.03 points below its actual teacher. Distillation cannot exceed its own teacher: that embedding is the target it fits. Trading 3 points for a model that fits in a browser is the point.

## How identification works

```
photo -> CLIP preprocess -> int8 ONNX tower -> 768-d embedding
      -> 11,167-way text classifier -> top 25 candidates
      -> score = sim/T + beta * log P(species | cell, month)
```

The geographic prior is empirical, built from 157,114,209 research-grade iNaturalist observations. The **month** axis is worth +1.2 points on its own (93.8 to 95.0, bootstrap 95% CI [+0.81, +1.69]): a Rufous Hummingbird in Seattle is plausible in July and not in January.

BirdLife range data is gone from scoring. It was measured **mathematically inert**: the fitted smoothing weight lands near 1e-10, ten orders of magnitude below the occurrence counts.

## What the app loses, and what replaces it

GPT returned three things a classifier cannot produce:

- **"No bird found"** was inferred from an empty candidate list. A classifier always returns 25 ranked species, so that branch would have become dead code. It now reads post-rerank confidence, a **better** signal: at threshold 0.7 it keeps 94.9% of photos at 97.91% accuracy, so the app interrupts on ~5% of uploads instead of 23-29%.
- **`multipleBirds`** is dropped. Nothing in a classifier counts birds, and the user knows about a second bird better than a threshold does.
- **`cropBox` auto-preview** is dropped. The manual crop UX covers it and is model-agnostic.

The crop prompt fires **at most once per photo**. Confidence tracks species ambiguity, not framing (Pearson 0.051 against relative bird area), so a naive re-prompt would loop forever.

**Anonymous users can now identify birds.** They were blocked with a 403 `Account required` because each call cost money. On-device inference removes the cost, so the gate has nothing left to protect.

## The 61.66 MiB download

| asset | size |
|---|---|
| `wingclip_visual_int8.onnx` | 13.72 MiB |
| `wingclip_visual_int8.data` | 24.00 MiB |
| `text_classifier_int8.bin` | 8.22 MiB |
| `occurrence.1fb61779.bin.gz` | 15.71 MiB |

Every file is under the **25 MiB Workers per-file cap**, which drove real design choices: the tower is split into graph plus external data, and the classifier is int8 with per-row scales. `scripts/check-asset-cap.mjs` now fails the build if anything exceeds it, after the ONNX runtime silently did exactly that.

The prior is named by **content hash**, not schema version, because `_headers` serves these immutable for a year and a fixed name would hand a stale blob to every existing user after a rebuild.

Download happens **on first identify**, never at page load, behind an explicit screen, cached persistently. Browsing a life list downloads nothing.

## What got deleted

The GPT server path is unreachable once the client stops calling it, so it goes:

`functions/api/identify-bird.ts`, `lib/bird-id.ts`, `lib/range-filter.ts`, `lib/bird-id-prompt`, `lib/bird-id-helpers`, `lib/ai-rate-limit.ts` (identify-bird was the only AI endpoint), `lib/http-error.ts`, plus their tests, the `identify-bird` path and two schemas from `openapi.yaml`, and both R2 and both AI bindings in `wrangler.toml`.

**Removing the `remote = true` bindings has a side benefit:** `wrangler dev` no longer authenticates against Cloudflare, so the e2e suite runs on a logged-out machine. That was previously impossible and made local e2e unrunnable.

`functions/lib/range-adjust.js` becomes `src/lib/equal-earth.ts`, 217 lines to 41. The BirdLife trust half was inert; the Equal Earth geometry stays because the occurrence prior is keyed by cell. It was `.js` under `functions/` because Node build scripts imported it directly and there is no `allowJs`; those scripts are gone, so it is client code now. Typing it caught a latent contract the untyped file hid: `xyToCell` returns null outside the grid.

## Verification

- **667 unit tests**, **43/43 e2e**, clean build, asset-cap OK
- Verified in a **clean shallow clone** (`git clone --depth 1 --single-branch`), not just a dev tree
- e2e runs **real on-device inference, no mocks**, asserting real species. Safe because weights are committed and inference is bit-deterministic: three runs of the same input give identical floats to the last decimal
- Preprocessing tests are **mutation-checked**: reintroducing `Math.round` for `bankersRound` fails exactly the half-to-even case and nothing else
- ONNX export is bit-exact against PyTorch, worst cosine 1.00000000

**One pre-existing bug fixed:** `playwright.config.ts` now uses `workers: 1`. Every spec shares one local D1 and the same dex, and several seed or clear it, so parallel workers corrupted each other. It looked like flakiness (one failure per run, rotating between files) but was a data race. It predates this PR; real inference just made the upload test slow enough to lose consistently.

## Context for whoever picks this up

Full experiment log, all 65 tracked decisions with findings, lives in `ml/README.md` on the `bioclip-distill` branch (PR #278). Every number above is traceable to a row there. Useful anchors: **G1** runtime choice, **G6** replacing GPT detection/framing, **G17** int8 classifier, **G19** the BirdLife retirement, **F14** the head-to-head, **F15** why BirdLife is inert.

### Next steps

- [x] **Try the preview on real photos.** Everything so far is Node or headless Chromium against local files. Real EXIF, real camera output, and a phone are what the preview exists to close.
- [x] **Publish WingCLIP-0.3 to Hugging Face.** Weights stay out of the repo (`ml/distill/runs` is 52 GB); only int8 inference artifacts ship.
- [x] **G18: Core ML export.** `coremltools` ships Linux wheels, so conversion runs on the training box; macOS is needed only to run prediction for verification. Preprocessing is the risk, not conversion: the web client hit three separate off-by-one bugs against PIL, each shifting the crop one pixel while every dimension still looked right.
- [x] **G20: unify the absence floor**, measured at zero cost.
- [x] **Port iOS to on-device ID.** `ios/WingDex/Services/DataService.swift:192` and two calls in `AddPhotosViewModel` hit the deleted `/api/identify-bird`. Merging this **breaks identification for iOS 0.7.3 on TestFlight**, which is acceptable only because that is two users, both the maintainer. iOS needs Core ML inference, the confidence gate, and the geo/month prior.

### Known gaps

- **Confidence trusts the prior.** For a genuine vagrant the model will be confidently wrong, which is exactly when an honest "not sure" matters most. Rare by definition, but real.

### CODE REVIEW

**Review by jlian — COMMENTED** [review]
On device ID seems to work in browser. UX issues from testing

- Progress bar for the model download a bit weird, seems to not fill up and suddenly it's done
- Progress bar and artificial blocking for the ID I think no longer makes sense, because each device would do it at different speed, so we might just change it to a spinner.
- Check if the 100% / 0% splits are real? Might want to log the odds before/after reranking. Maybe it makes more sense to not try to have the values add up to 100%? It's not probability, more like confidence level.

Image

- Reconsider abstention threshold (pic is of a dog running) 

image
